evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -170,14 +144,22 @@ def fit(self, X, y=None):
         Raises:
             ValueError: If X was passed to `fit` but not passed in `predict`.
         """
+        X, y = self._manage_woodwork(X, y)
         if y is None:
             raise ValueError("ARIMA Regressor requires y as input.")
+        if X is not None:
```

Comment:
This addresses #2960. The issue is that since `TimeSeriesFeaturizer` isn't included in ARIMA-based pipelines, the potential for target leakage can occur.

Reply:
I think we should back this out of this PR and tackle it in it's own issue because I don't think this is the best fix for the problem.

I think the root cause of the problem is including features that will not be known during prediction time during fit. And whether or not a feature is known-in-advanced is independent of it's correlation to the target (which is what TargetLeakeage is checking). It's possible that there is a feature that will not be known in advanced that is not picked up by TargetLeakage and it's also possible that a feature that will be known in advanced will be dropped because it's correlated with the target. 

I think we gotta think this through a bit more it's possible #2960 is blocked by #2511